src/fvm_multicell.hpp

```diff
@@ -85,8 +85,56 @@ class fvm_multicell {
         return (this->*h.first)[h.second];
     }
 
-    /// integrate all cell state forward in time
-    void advance(double dt);
+    /// Initialize state prior to a sequence of integration steps.
+    void start_integration(value_type tfinal, value_type dt_max) {
+        EXPECTS(tfinal>t_);
+        EXPECTS(dt_max>0);
+
+        tfinal_ = tfinal;
+        dt_max_ = dt_max;
+        integration_running_ = true;
+
+        // TODO: Placeholder; construct backend event delivery
+        // data structure from events added.
+        EXPECTS(events_.empty());
+        for (auto& ev: staged_events_) {
+            EXPECTS(ev.time<tfinal);
+            events_.push(std::move(ev));
+        }
+
+        staged_events_.clear();
+    }
+
+    /// Advance one integration step.
```

Comment:
I didn't think it was really that transparent — didn't want to use `advance` in the method name to avoid confusion with the previous method and semantics, and I thought the name in isolation was a little ambiguous.